### ISSUE TITLE
Question: Will the optional apply when used with logic operators?

### ISSUE THREAD

**Issue by dreinon** [tpaviot/ProcessScheduler#12]
I'm using the following structure:
```python
for i in range(num_practices_double):
        practice_num += 1
        students_tasks.append(ps.FixedDurationTask(f'{stud_id}__{practice_num}', duration=2, priority = (21 - i),optional=True))
        students_tasks[-1].add_required_resource(Car)
        av_list = []
        for j in availability:
            av_list.append(ps.and_([ps.TaskStartAfterLax(students_tasks[-1], j[0],optional=True), ps.TaskEndBeforeLax(students_tasks[-1], j[1],optional=True)]))
        constraints.append(ps.ForceApplyNOptionalConstraints(av_list,1))
```

What I want to be optional is the constraint made by the and operator, I mean, the options would be, for example:
- Start at n1 AND end at n2
- Start at n3 AND and at n6

So I imagine something like ps.and_(logic, optional=true).

Thank you again :)

**Comment by tpaviot** [comment]
hmm, I don't know :smile: Just try, it might work

**Comment by dreinon** [comment]
Solved in issue #13
